src/lints/enum_variant_field_deprecated.ron

```diff
@@ -0,0 +1,66 @@
+SemverQuery(
+    id: "enum_variant_field_deprecated",
+    human_readable_name: "enum variant field marked with #[deprecated]",
+    description: "A field in an enum variant was marked with the #[deprecated] attribute.",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                        variant {
+                            ... on StructVariant {
+                                variant_name: name @output
+                                field {
+                                    field_name: name @output @tag
```

Comment:
Both variants and their fields can be `#[doc(hidden)]`, so we'll need `public_api_eligible` checks in both the variant and the field blocks, in both baseline and current. Test cases for all these situations would be good too.